src/orchard/print.clj

```diff
@@ -43,8 +42,18 @@
   "Maximum total size of the resulting string."
   Integer/MAX_VALUE)
 
+(defn- print-coll-item
+  "Print an item in the context of a collection. When printing a map, don't print
+  `[]` characters around map entries."
+  [^TruncatingStringWriter w, x, map?]
+  (if (and map? (instance? Map$Entry x))
+    (do (print (.getKey ^Map$Entry x) w)
+        (.write w " ")
+        (print (.getValue ^Map$Entry x) w))
+    (print x w)))
+
 (defn- print-coll [^TruncatingStringWriter w, ^Iterable x, ^String sep
-                   ^String prefix, ^String suffix]
+                   ^String prefix, ^String suffix, map?]
```

Comment:
Not a big fan of this boolean param, but I can live with it. Perhaps we can at least make it somewhat optional, so we don't have to pass `false` around.

Reply:
Updated it to be optional. Does it look that much better for you? It's a private function anyway.

Reply:
Not much better, but a bit better - in general I'm not a big fan of boolean params, as I always have to consult the method signature to remember/understand what some boolean flag is supposed to do. But I agree it's not a big deal in our case, given it's a private API.